goschedstats: cleanup and minor API improvement

RaduBerinde opened the pull request:
This set of changes improves the goschedstats API discoverability and readability. There should be no functional changes.

Epic: None
Release note: None

#### goschedstats: rename runnable.go to runnable_enabled.go


#### goschedstats: rename runnable_common.go to runnable.go


#### goschedstats: refactor code

We now have a single instance of the public APIs.

We also move the definition of the cluster setting so that it exists
in all builds (even if it doesn't do anything, a server built without
our runtime should recognize the setting name).

#### goschedstats: minor API improvement

`RegisterRunnableCountCallback` now returns an `ok bool`, which makes
it more evident that it can fail.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/142599)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1, 2 of 2 files at r2, 1 of 3 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142599#-:-OL1c9M25hu1SvNmsCCN:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde)

___
*[`pkg/server/server.go` line 599 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/142599#-OL1bji08jxGCL6O-o-e:-OL1bji08jxGCL6O-o-f:b50ve40) ([raw file](https://github.com/cockroachdb/cockroach/blob/6a7abe29c3a0714ffa9370937f3a8f92ee5e2896/pkg/server/server.go#L599)):*
> ```Go
> 			goschedstats.UnregisterRunnableCountCallback(cbID)
> 		}))
> 	}
> ```

should we `log.Warning` for the `!ok` case?
___
*[`pkg/util/goschedstats/runnable.go` line 16 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/142599#-OL1bsg-7AZdw45Z8G5q:-OL1bsg03pY5ujNBYBEm:b7in0e7) ([raw file](https://github.com/cockroachdb/cockroach/blob/6a7abe29c3a0714ffa9370937f3a8f92ee5e2896/pkg/util/goschedstats/runnable.go#L16)):*
> ```Go
> 
> // Enabled is true if the system has access to the internal goroutine
> // statistics (i.e. if we were built using our Go fork).
> ```

nit: if CockroachDB was built ...


<!-- Sent from Reviewable.io -->

RaduBerinde committed:
goschedstats: rename runnable.go to runnable_enabled.go

RaduBerinde committed:
goschedstats: refactor code

We now have a single instance of the public APIs.

We also move the definition of the cluster setting so that it exists
in all builds (even if it doesn't do anything, a server built without
our runtime should recognize the setting name).

RaduBerinde committed:
goschedstats: rename runnable_common.go to runnable.go

RaduBerinde committed:
goschedstats: minor API improvement

`RegisterRunnableCountCallback` now returns an `ok bool`, which makes
it more evident that it can fail.

RaduBerinde reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/142599#-:-OL4c8Tv0Nw51gUrhTPn:b-814g8k)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[`pkg/server/server.go` line 599 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/142599#-OL1bji08jxGCL6O-o-e:-OL4c9O_3kchPqm4kkJ4:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6a7abe29c3a0714ffa9370937f3a8f92ee5e2896/pkg/server/server.go#L599)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

should we `log.Warning` for the `!ok` case?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->
